cynic-codegen/src/enum_derive/input.rs

```diff
@@ -24,3 +27,23 @@ pub struct EnumDeriveVariant {
     #[darling(default)]
     pub(super) rename: Option<SpannedValue<String>>,
 }
+
+impl EnumDeriveInput {
+    pub fn graphql_type_name(&self) -> String {
+        // May be have a good way.
+        String::from(
+            &*(self
+                .graphql_type
+                .as_ref()
+                .map(|val| val.clone())
+                .unwrap_or(SpannedValue::from(self.ident.to_string()))),
+        )
```

Comment:
I think if you convert to a string while still in the `Option` this might be nicer:

```suggestion
          self
              .graphql_type
              .as_ref()
              .map(String::from)
              .unwrap_or_else(|| self.ident.to_string()),
```

It's also better to use `unwrap_or_else` when you're calling a function inside the `unwrap_or` - that way you avoid calling that function if the `Option` is actually `Some`.

Reply:
@obmarg Maybe like here?
```rust
pub fn graphql_type_name(&self) -> String {
        self
            .graphql_type
            .as_ref()
            .map(|sp| String::from(&**sp))
            .unwrap_or_else(|| self.ident.to_string())
    }
```

Reply:
Yeah, that'd also do.  Possible that `map(|sp| sp.to_string())`would work without the `&**`?  Or maybe `map(|sp| (*sp).to_string())`?

Happy with what you have if either of those don't do the job though